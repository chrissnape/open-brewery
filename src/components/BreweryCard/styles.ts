import { StyleSheet } from 'react-native';
import { borderColour, borderWidth, primaryColour, padding, fontSize } from '../../utils/styles';

export default StyleSheet.create({
  card: {
    justifyContent: 'space-between',
    backgroundColor: '#fff',
    borderColor: borderColour,
    borderWidth,
    flexDirection: 'row',
    padding,
    width: '100%',
  },
  icon: {
    color: primaryColour,
    fontSize: 50,
  },
  title: {
    color: primaryColour,
    fontWeight: 'bold',
    fontSize: fontSize + 4,
    marginBottom: 5,
  },
});
