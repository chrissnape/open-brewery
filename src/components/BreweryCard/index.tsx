import React, { FC, Fragment } from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import styles from './styles';
import { Ionicons } from '@expo/vector-icons';

type Props = {
  city: string;
  name: string;
  onPress: () => void,
  state: string;
}

const BreweryCardComponent: FC <Props> = ({ city, name, onPress, state }): JSX.Element => (
  <TouchableWithoutFeedback onPress={onPress}>
    <View style={styles.card}>
      <View>
        <Text style={styles.title}>{name}</Text>
        <Text>{`${city}, ${state}`}</Text>
      </View>
      <Ionicons style={styles.icon} name="chevron-forward-outline" />
    </View>
   </TouchableWithoutFeedback>
);

export default BreweryCardComponent;
