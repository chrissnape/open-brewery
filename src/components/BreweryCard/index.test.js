import React from 'react';
import renderer from 'react-test-renderer';
import BreweryCard from './index';

const props = {
  city: 'Los Angeles',
  name: 'Lorem ipsum',
  state: 'California',
}

describe('<BreweryCard />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BreweryCard { ...props } />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
