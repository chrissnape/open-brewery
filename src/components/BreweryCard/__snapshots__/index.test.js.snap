// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreweryCard /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#ccc",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#065535",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Lorem ipsum
    </Text>
    <Text>
      Los Angeles, California
    </Text>
  </View>
  <Text />
</View>
`;
